docsource/use_cases/value_mapping.rst

```diff
@@ -0,0 +1,183 @@
+
+same table as in version 14.
+
+**Problem**:
+
+The old value does not correspond to anything in the new version
+and is therefore no longer displayed in the interface.
+
+.. image:: ./images/value-mapping-error.png
+
+V15 table mail_activity_type (With migration script)
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+.. csv-table::
+   :header: "id", "res_id", "res_model", "rating", "rating_text", "openupgrade_legacy_14_0_rating_text"
+
+   "1", "3", "project.task", "5", "top", "satisfied"
+   "2", "22", "project.task", "1", "ko", "highly_dissatisfied"
+   "3", "4", "project.task", "5", "top", "satisfied"
+   "4", "2", "project.task", "1", "ko", "highly_dissatisfied"
+
+Contribution to OpenUpgrade
+---------------------------
+
+Update ``upgrade_analysis_work.txt`` file
+"""""""""""""""""""""""""""""""""""""""""
+
+* Mention the operation performed, starting with ``# DONE:``
+
+.. code-block:: text
+
+    ---Fields in module 'rating'---
+    rating       / rating.rating            / rating_text (selection)       :
+    selection_keys is now '['ko', 'none', 'ok', 'top']'
+    ('['highly_dissatisfied', 'no_rating', 'not_satisfied', 'satisfied']')
+    # DONE: post-migration: mapped value from old keys to new keys
+
+See `Full V15 Work Analysis File <https://github.com/OCA/OpenUpgrade/blob/0d918b5b5dfb3125a5d881c8ccf4237997d8c7f5/openupgrade_scripts/scripts/rating/15.0.1.0/upgrade_analysis_work.txt#L3>`_.
+
+Write migration Script
+""""""""""""""""""""""
+
+In the ``pre-migration.py`` script add:
+
+.. code-block:: python
+
+    from openupgradelib import openupgrade
+
+    @openupgrade.migrate()
+    def migrate(env, version):
+        openupgrade.copy_columns(
+            env.cr,
+            {"rating_rating": [("rating_text", None, None)]},
+        )
+
+In the ``post-migration.py`` script add:
+
+.. code-block:: python
+
+    from openupgradelib import openupgrade
+
+    @openupgrade.migrate()
+    def migrate(env, version):
+        openupgrade.map_values(
+            env.cr,
+            openupgrade.get_legacy_name("rating_text"),
+            "rating_text",
+            [
+                ("satisfied", "top"),
+                ("not satisfied", "ok"),
+                ("highly_dissatisfied", "ko"),
+                ("no_rating", "none"),
+            ],
+            table="rating_rating",
+        )
+
+
+See `Full pre migration Script <https://github.com/OCA/OpenUpgrade/blob/0d918b5b5dfb3125a5d881c8ccf4237997d8c7f5/openupgrade_scripts/scripts/rating/15.0.1.0/post-migration.py#L1>`_.
+
+
+Notes
+-----
+
+* Sometime, there is a loss of information.
+  that is the case in the above example:
+  Some mapping are correct:
+
+    * ``satisfied`` -> ``top``
+    * ``highly_dissatisfied`` -> ``ko``
+    * ``no_rating`` -> ``none``
+
+  But the last mapping is ``not satisfied`` -> ``ok``, which changes the semantics of the data.
+  It is not always possible to do otherwise and the developer must make arbitrary choices.
```

Comment:
```suggestion
  It is not always possible to do otherwise and the developer must make the best suited choices.
```

Reply:
This seems to be reverted

Reply:
yes, by mistake, I had to run some  ``git mistake --lose-github-suggestions``. 
thanks ! 
Fixed.